firebase-config/src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigTest.java

```diff
@@ -1039,6 +1043,45 @@ public void personalization_hasMetadata_successful_without_analytics() throws Ex
             });
   }
 
+  @Test
+  public void realtime_addListener_success() {
+    ConfigUpdateListener eventListener = generateEmptyRealtimeListener();
+    when(mockRealtimeHttpClient.addRealtimeConfigUpdateListener(eventListener))
+        .thenReturn(
+            new ConfigUpdateListenerRegistration() {
+              @Override
+              public void remove() {}
+            });
+    ConfigUpdateListenerRegistration registration =
+        mockRealtimeHttpClient.addRealtimeConfigUpdateListener(eventListener);
+    verify(mockRealtimeHttpClient).addRealtimeConfigUpdateListener(eventListener);
+    assertThat(registration).isNotNull();
+  }
+
+  @Test
+  public void realtime_addListener_fail() {
+    when(mockRealtimeHttpClient.addRealtimeConfigUpdateListener(null)).thenReturn(null);
+
+    ConfigUpdateListenerRegistration registration =
+        mockRealtimeHttpClient.addRealtimeConfigUpdateListener(null);
```

Comment:
Should we throw instead of returning null?

Reply:
Actually updated the method to have NonNull annotations similar to firestore(https://github.com/firebase/firebase-android-sdk/blob/946624278814ec6d467dd145f45421348e740225/firebase-firestore/src/main/java/com/google/firebase/firestore/DocumentReference.java#L366). 
Went ahead and removed the bad listener test too.